src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -3925,6 +3940,121 @@ void EmitStackPop()
             }
         }
 
+        protected void EmitScan(DynamicMethod findFirstCharMethod, DynamicMethod goMethod)
+        {
+            LocalBuilder bump = DeclareInt32();
+            LocalBuilder stoppos = DeclareInt32();
+            Label returnLabel = DefineLabel();
+
+            Label notRightToLeft = DefineLabel();
+            // int bump = 1
+            Ldc(1);
+            Stloc(bump);
+
+            // int stoppos = text.Length
+            _ilg!.Emit(OpCodes.Ldarga_S, 1);
+            Call(s_spanGetLengthMethod);
+            Stloc(stoppos);
+
+            // if (runregex.RightToLeft)
+            // {
+            Ldthisfld(s_runregexField);
+            Callvirt(s_regexGetRightToLeft);
+            BrfalseFar(notRightToLeft);
+
+            // bump = -1;
+            // stoppos = 0;
+            // }
+            Ldc(-1);
+            Stloc(bump);
+            Ldc(0);
+            Stloc(stoppos);
+            MarkLabel(notRightToLeft);
+
+            // while (true)
+            Label whileLoopEnd = DefineLabel();
+            Label whileLoopBody = DefineLabel();
+            MarkLabel(whileLoopBody);
+
+            // if (FindFirstChar(text))
+            Label afterFindFirstCharLabel = DefineLabel();
+            Ldthis();
+            Ldarg_1();
+            Call(findFirstCharMethod);
+            BrfalseFar(afterFindFirstCharLabel);
+
+            // CheckTimeout();
+            Ldthis();
+            Call(s_checkTimeoutMethod);
```

Comment:
Missing guard on the timeout check.

Reply:
Do you mind elaborating? Are you referring to the existing `if (ignoreTimeout)`? If so, we don't do that any longer since timeout doesn't get passed through to Scan. Before we had the guard because we were calling `DoCheckTimeout()` as opposed to `CheckTimeout` which has the guard inside of it.

Reply:
> Do you mind elaborating?

```C#
if (_hasTimeout)
{
    Ldthis();
    Call(s_checkTimeoutMethod);
}
```